core/server/src/test/java/alluxio/worker/file/UnderFileSystemManagerTest.java

```diff
@@ -303,6 +304,35 @@ public void getInputStreamAtPositionTest() throws Exception {
   }
 
   /**
+   * Tests getting an input stream to a valid file at a position returns the correct input stream
+   * when using a UFS which supports opening at a position.
+   */
+  // TODO(calvin): Generalize this when openAtPosition is part of an interface
+  @Test
+  public void getInputStreamAtPositionOptimizedTest() throws Exception {
+    S3AUnderFileSystem ufs = Mockito.mock(S3AUnderFileSystem.class);
+    Mockito.when(ufs.create(Mockito.anyString())).thenReturn(mMockOutputStream);
+    Mockito.when(ufs.create(Mockito.anyString(),
+        Mockito.any(CreateOptions.class))).thenReturn(mMockOutputStream);
+    Mockito.when(ufs.openAtPosition(Mockito.anyString(), Mockito.anyLong())).thenReturn(
+        mMockInputStream);
+    Mockito.when(ufs.rename(Mockito.anyString(), Mockito.anyString())).thenReturn(true);
+    Mockito.when(ufs.getFileSize(Mockito.anyString())).thenReturn(FILE_LENGTH);
+    PowerMockito.mockStatic(UnderFileSystem.class);
+    Mockito.when(UnderFileSystem.get(Mockito.anyString())).thenReturn(ufs);
+
+    long position = FILE_LENGTH - 1;
+    Mockito.when(ufs.exists(mUri.toString())).thenReturn(true);
+    long id = mManager.openFile(SESSION_ID, mUri);
+    Mockito.when(mMockInputStream.read()).thenReturn(5);
+    InputStream in = mManager.getInputStreamAtPosition(id, position);
+    Assert.assertEquals(5, in.read());
+    // Due to the optimization, we should not need to skip
+    Mockito.verify(mMockInputStream, Mockito.never()).skip(position);
```

Comment:
I agree, do we want to address this as we modify unit tests or have a ticket for it across the codebase?